src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectOrCollectionInitializerExpression.cs

```diff
@@ -394,9 +395,8 @@ private BoundExpression VisitObjectInitializerMember(BoundObjectInitializerMembe
 
                 case BoundKind.ArrayAccess:
                     {
-                        Debug.Assert(rewrittenLeft is { });
-                        var arrayAccess = (BoundArrayAccess)rewrittenLeft;
-
+                        var arrayAccess = (BoundArrayAccess)VisitExpression(left);
```

Comment:
>VisitExpression [](http://example.com/codeflow?start=60&length=15)

```VisitArrayAccess```? BTW, it looks like ```VisitArrayAccess``` has handling for ```System.Range``` used as an index. Is that handling still necessary or can be removed? Is it possible to run into it on this code path? It looks like we will not get ```BoundArrayAccess``` back in that case.

Reply:
Thanks for catching this, there was a bug here with Range in array access.